odoo/addons/base/tests/test_module.py

```diff
@@ -0,0 +1,137 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import os.path
+import tempfile
+from os.path import join as opj
+from unittest.mock import patch
+
+import odoo.addons
+from odoo.modules.module import load_manifest
+from odoo.release import major_version
+from odoo.tests.common import BaseCase
+
+
+class TestModuleManifest(BaseCase):
+    @classmethod
+    def setUpClass(cls):
+        load_manifest.cache_clear()
+
+        cls._tmp_dir = tempfile.TemporaryDirectory(prefix='odoo-test-addons-')
+        cls.addons_path = cls._tmp_dir.name
+
+        patcher = patch.object(odoo.addons, '__path__', [cls.addons_path])
+        patcher.start()
+        cls.addClassCleanup(patcher.stop)
+
+    @classmethod
+    def tearDownClass(cls):
+        load_manifest.cache_clear()
+        cls._tmp_dir.cleanup()
```

Comment:
tearDown methods suck IMHO, as they are not executed when the corresponding setUp method fails.
The good practice is to add a `self.addCleanup()` or `cls.addClassCleanup()` right after the corresponding patch.